### PR TITLE
Update Cron, Timer, and Doser

### LINKED ISSUE #830: Add UI support for new timer API
3.0 timer api has been updated to include three different types of targe
- reminder, equipment or macro
- The core scheduling fields (second, minute, hour, day) has been extended with month and day of the week. Here is an example of new timer with a equipment target
- The target field now encapsulate all time type specific information. For equipment it looks like this
```json
{
  "id": "2",
  "name": "Pump 1",
  "enable": true,
  "type": "equipment",
  "month": "*",
  "week": "?",
  "day": "*",
  "hour": "*",
  "minute": "*",
  "second": "1,31",
  "target": {
    "id": "2",
    "revert": true,
    "on": true,
    "duration": 15
  }
}
```
- For macro target 
```json
{
"id": "2",
  "name": "Pump 1",
  "enable": true,
  "type": "macrot",
  "month": "*",
  "week": "?",
  "day": "*",
  "hour": "*",
  "minute": "*",
  "second": "1,31",
   "target": {
     "id": "1"
   }
}
```
- for reminder target
- For macro target 
```json
{
"id": "2",
  "name": "Notification",
  "enable": true,
  "type": "reminder",
  "month": "*",
  "week": "?",
  "day": "*",
  "hour": "12",
  "minute": "1",
  "second": "1,",
   "target": {
     "title": "test"
      "message": "test"
   }
}
```

The UI should show drop down for choosing the specific macros from defined ones when timer type is selected as macro.

### PR DESCRIPTION
**Pull request by Ranthalion**
## Description

Update timer UI to work with new Cron format and targets.

## Related Issue

Fixes #830

### CODE REVIEW

File: controller/daemon/settings.go [hunk 1]
```diff
@@ -16,7 +17,7 @@ func loadSettings(store storage.Store) (settings.Settings, error) {
 }
 
 func initializeSettings(store storage.Store) (settings.Settings, error) {
-	if os.Getenv("DEV_MODE") == "1" {
+	if os.Getenv("DEV_MODE") == "1" || runtime.GOOS == "Windows" {
```
Comment (ranjib) [inline controller/daemon/settings.go]: :sweet_potato:

File: front-end/src/timers/edit_timer.jsx [hunk 2]
```diff
@@ -35,6 +35,40 @@ const EditTimer = ({
...
+      target: {
+        name: 'target',
+        value: targetFor(e.target.value)
+      }
+    }
+
+    // allow the original change of the type field to proceed
+    handleChange(e, props)
+    // notify the change of the target
+    handleChange(event, props)
```
Comment (ranjib) [inline front-end/src/timers/edit_timer.jsx]: :volcano:

File: front-end/src/timers/target.jsx [hunk 3]
```diff
@@ -182,13 +169,13 @@ const Target = (props) => {
...
-              disabled={readOnly}
+              name={NameFor(props.name, 'message')}
+              disabled={props.readOnly}
               className={classNames('form-control', {
-                'is-invalid': ShowError(NameFor(name, 'message'), touched, errors)
+                'is-invalid': ShowError(NameFor(props.name, 'message'), props.touched, props.errors)
               })}
             />
-            <ErrorFor {...props} name={NameFor(name, 'message')} />
+            <ErrorFor {...props} name={NameFor(props.name, 'message')} />
```
Comment (ranjib) [inline front-end/src/timers/target.jsx]: i was close :-) , but still dont know what im doing

**Comment by ranjib** [comment]
:shipit:  Let's wait for travis to go green. I'll merge soon after that

File: front-end/src/timers/target.jsx (hunk 3 above)
Comment (Ranthalion) [reply front-end/src/timers/target.jsx]: Ah.  That probably wasn't necessary.  I can change back to using let if you would like. Root cause was that Formik didn't know the shape of the data.

**Pull request merged**